src/Stratis.Bitcoin.Features.Wallet/Wallet.cs

```diff
@@ -13,6 +13,12 @@ namespace Stratis.Bitcoin.Features.Wallet
     /// </summary>
     public class Wallet
     {
+        /// <summary>Account numbers greater or equal to this number are reserved for special purpose account indexes.</summary>
+        public const int SpecialPurposeAccountIndexesStart = 100_000_000;
+
+        /// <summary>Filter for identifying normal wallet accounts.</summary>
+        public static Func<HdAccount, bool> NormalAccounts = a => a.Index < SpecialPurposeAccountIndexesStart;
```

Comment:
Instead of having this, can we have a method `GetSpecialPurposeAccountsByCoinType`?
Seems to be simpler.

Reply:
`GetSpecialPurposeAccountsByCoinType` may be a bit vague. Also there are 3 different subsets of accounts we are interested in:
- All the accounts - used by `TransactionFoundInternal` and `LoadKeysLookupLock`.
- Normal accounts - to support default account selection.
- Normal accounts plus the hot wallet accounts - to get UTXO's for staking.

Here are the various ways we are passing flags down the call  hierarchy:

```
GetAddressBalance(string address)
-> GetAllAddressesByCoinType(CoinType coinType, Func<HdAccount, bool> accountFilter = null /* Normal Accounts */)
   -> GetAccountsByCoinType(CoinType coinType, Func<HdAccount, bool> accountFilter = null)

LoadKeysLookup
-> GetAllAddressesByCoinType(CoinType coinType, Func<HdAccount, bool> accountFilter = null /* ALL Accounts */)
   -> GetAccountsByCoinType(CoinType coinType, Func<HdAccount, bool> accountFilter = null)

TransactionFoundInternal(Script script)
-> GetAccountsByCoinType(CoinType coinType, Func<HdAccount, bool> accountFilter = null /* ALL Accounts */)

GetAllTransactionsByCoinType(CoinType coinType)
GetAllPubKeysByCoinType(CoinType coinType)
Start()
GetHistory(string walletName, string accountName = null) (when null)
GetBalances(string walletName, string accountName = null) (when null)
GetAccounts(string walletName)
RemoveTransactionsByIdsLocked(string walletName, IEnumerable<uint256> transactionsIds)
RemoveAllTransactions(string walletName)
-> GetAccountsByCoinType(CoinType coinType, Func<HdAccount, bool> accountFilter = null /* Normal Accounts */)

GetSpendableTransactionsInWalletForStaking(string walletName, int confirmations = 0)
-> GetSpendableTransactionsInWallet(string walletName, int confirmations, Func<HdAccount, bool> accountFilter /* Normal + Hot Wallet Accounts */)
  -> GetAllSpendableTransactions(CoinType coinType, int currentChainHeight, int confirmations = 0, Func<HdAccount, bool> accountFilter = null)
    -> GetAccountsByCoinType(CoinType coinType, Func<HdAccount, bool> accountFilter = null)

GetSpendableTransactionsInWallet(string walletName, int confirmations = 0)
-> GetSpendableTransactionsInWallet(string walletName, int confirmations, Func<HdAccount, bool> accountFilter /* Normal Accounts */)
  -> GetAllSpendableTransactions(CoinType coinType, int currentChainHeight, int confirmations = 0, Func<HdAccount, bool> accountFilter = null)
    -> GetAccountsByCoinType(CoinType coinType, Func<HdAccount, bool> accountFilter = null)
```